myhoard/backup_stream.py

```diff
@@ -257,6 +257,21 @@ def binlog_upload_age(self):
             # If we're not supposed to be streaming binlogs then report age as 0
             return 0
 
+    @property
+    def binlog_upload_delay(self) -> int:
+        """Returns the number of seconds since the oldest pending binlog was written."""
+        mtimes = set()
+        for binlog in self.state["pending_binlogs"]:
+            try:
+                mtimes.add(os.stat(binlog["full_name"]).st_mtime)
```

Comment:
why not just
```
min_mtime = min(os.stat(binlog["full_name"]).st_mtime, min_mtime)
```
and use it later

Reply:
This is in a `try` .. `except` block as the file may no longer be present.

Reply:
I mean there is no point in keeping all mtimes in a set, when you need only one value.

Reply:
Oh I see. Sure, I will make that change. Although this is a local variable, it's good practice to bound memory usage.